ui/App/SharedComponents/UserIdentity.svelte

```diff
@@ -0,0 +1,152 @@
+<!--
+ Copyright © 2021 The Radicle Upstream Contributors
+
+ This file is part of radicle-upstream, distributed under the GPLv3
+ with Radicle Linking Exception. For full terms see the included
+ LICENSE file.
+-->
+<script lang="ts">
+  import type * as proxyIdentity from "ui/src/proxy/identity";
+
+  import * as error from "ui/src/error";
+  import * as modal from "ui/src/modal";
+  import * as proxy from "ui/src/proxy";
+  import * as remote from "ui/src/remote";
+  import * as router from "ui/src/router";
+  import * as session from "ui/src/session";
+  import * as svelteStore from "ui/src/svelteStore";
+
+  import { Avatar, CopyableIdentifier, Hovercard } from "ui/DesignSystem";
+
+  export let urn: string;
+  export let handle: string | undefined = undefined;
+  export let disableHovercard: boolean = false;
+  export let modalStyle: string = "top: -2rem; left: -17rem;";
+  export let boldHandle: boolean = false;
+
+  const userStore = remote.createStore<proxyIdentity.RemoteIdentity>();
+
+  async function fetchUser(urn: string) {
+    const userStore_ = svelteStore.get(userStore);
+    if (
+      userStore_.status === remote.Status.Success &&
+      userStore_.data.urn === urn
+    ) {
+      // Use cached data if we already have it.
+      return;
+    }
```

Comment:
What you are saying and what you have implemented makes total sense. I was considering the following scenario:

1. The component is rendered with URN A
2. The user hovers the component
3. The component displays the hovercard with information about person A after it is loaded
4. The `urn` property of the component changes to B
5. The user hovers the component
6. The component displays the hovercard with information about person _A while loading_ information about person B.
7. After the info about person B is loaded it is swapped into the hovercard.

Clearly 6. is an issue because we show outdated information to the user. What I suggested is to clear out the cache if `urn` has changed so that we don’t display that outdated information.

I believe this is a theoretical scenario at the moment because 4. does seem to occur anywhere in the app. But in the future we might have a situation where this is the case. Since the fix is minor I’d opt for adding it to avoid headaches in the future.